OpenRA.Game/Traits/World/ActorMap.cs

```diff
@@ -376,6 +376,11 @@ static void RemoveInfluenceInner(ref InfluenceNode influenceNode, Actor toRemove
 
 		public void Tick(Actor self)
 		{
+			UpdateTriggers();
+		}
+
+		public void UpdateTriggers()
```

Comment:
You're missing the crucial issue here: by the time the Tick runs, the old owner data is lost.